File: nomor 2/golang/ex12.go
package main

import (
	"fmt"
	"strconv"
)

func generateData(n int) []string {
	baseNumber := "882"
	var customers []string
	var mobileNumber string
	for i := 0; i < n; i++ {
		mobileNumber = baseNumber
		mobileNumber = fmt.Sprintf("%s%09s", mobileNumber, "0")
		mobileNumber += strconv.Itoa(i)
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(input []string) {
	for i := 0; i < len(input); i++ {
		fmt.Println("Sending Promo to", input[i])
	}
	fmt.Println("Its Finished to send Promo to", len(input), "Customers")
	for i := 0; i < len(input); i++ {
		fmt.Println("Sending Discount to", input[i])
	}
	fmt.Println("Its Finished to send Discount to", len(input), "Customers")
}

func main() {
	data := generateData(1000)
	sendPromoDiscount(data)
}

File: nomor 2/kotlin/ex12.kt
fun generateData(n: Int): List<String> {
    val baseNumber = "882"
    val customers = mutableListOf<String>()
    for (i in 0 until n) {
        val mobileNumber = "$baseNumber${"0".repeat(9)}$i".takeLast(10)
        customers.add(mobileNumber)
    }
    return customers
}

fun sendPromoDiscount(input: List<String>) {
    for (mobileNumber in input) {
        println("Sending Promo to $mobileNumber")
    }
    println("Its Finished to send Promo to ${input.size} Customers")
    for (mobileNumber in input) {
        println("Sending Discount to $mobileNumber")
    }
    println("Its Finished to send Discount to ${input.size} Customers")
}

fun main() {
    val data = generateData(10)
    sendPromoDiscount(data)
}
